include/asterisk/app.h

```diff
@@ -123,6 +123,25 @@ enum ast_timelen {
 */
 int ast_ivr_menu_run(struct ast_channel *c, struct ast_ivr_menu *menu, void *cbdata);
 
+/*! \brief Separates a filename list honoring escaped ampersands.
+ *
+ * Given a list of ampersand delimited filenames (\c one&two&three),
+ * progressively returns each filename while allowing the ampersand to be
+ * escaped with a backslash.
+ *
+ * Similar in behavior to <tt>strsep(&data, "&")</tt> except:
+ *   - Does not return empty elements
+ *   - Removes the backslash character from the returned string if it is
+ *     escaping an ampersand
+ *
+ * \param stringp A pointer to the \c \0 terminated string to traverse.
+ *                Like \c strsep, the \c stringp argument is modified by
+ *                this function.
+ * \return A non-\c NULL pointer to the next filename from \c stringp,
+ *         or \c NULL if there are no remaining filenames.
+ */
+char *ast_app_filename_list_iterate(char **restrict stringp);
```

Comment:
I'm starting to think that it might be better to make this function more generic and take the separator as an argument. Basically a `strsep(...)` that allows you to escape the separator and doesn't return empty elements.